test/models/deploy_test.rb

```diff
@@ -241,6 +255,32 @@ def deploy!
     end
   end
 
+  describe "to_csv associations with_deleted" do
+    let(:deployer) { users(:super_admin) }
+    let(:other_user) { users(:deployer_buddy) }
+    let(:prod) { stages(:test_production) }
+    let(:prod_deploy) { deploys(:succeeded_production_test) }
+    let(:job) { jobs(:succeeded_production_test) }
+
+    before { Stage.any_instance.stubs(:deploy_requires_approval?).returns true }
+
+    it "returns array with deleted object values" do
+      project.update_attributes(repository_url: "https://github.com/samson-test-org/example-project")
```

Comment:
Had to do this to get it to pass the test.  This probably means that because of the validations in projects it will fail if the project is soft deleted and the repo is deleted.

Not sure if its possible to preload all the associated data into the object without associations where it skips any validations for reporting.